src/main/java/com/znsio/api/VisualTest.java

```diff
@@ -0,0 +1,123 @@
+            ApplitoolsConfigurationManager.setUFGBrowserConfig(eyesConfig);
+        }
+        runner = ApplitoolsConfigurationManager.isUltraFastGridEnabled() ? new VisualGridRunner(new RunnerOptions()
+                .testConcurrency(ApplitoolsConfigurationManager.getConcurrency())) : new ClassicRunner();
+        runner.setDontCloseBatches(true);
+    }
+
+    @BeforeMethod
+    public void initiateVisualTests(Method method) {
+        LOGGER.info("@BeforeMethod of VisualTest called: " + method.getName());
+        eyes = new Eyes(runner);
+        if (ApplitoolsConfigurationManager.isLogsEnabled()) {
+            eyes.setLogHandler(new StdoutLogHandler(ApplitoolsConfigurationManager.isLogsEnabled()));
+        }
+        eyes.setConfiguration(eyesConfig);
+        boolean isTestPartOfVisualGroup = Arrays.toString(method.getAnnotation(Test.class).groups())
+                .contains(ApplitoolsConfigurationManager.getVisualValidationGroupName());
+        LOGGER.info("Is test part of '" +
+                ApplitoolsConfigurationManager.getVisualValidationGroupName() + "' group: " + isTestPartOfVisualGroup);
+        eyes.setIsDisabled((!isTestPartOfVisualGroup || ApplitoolsConfigurationManager.isApplitoolsDisabled()));
+        LOGGER.info("Is Applitools enabled for " + method.getName() + ": " + !eyes.getIsDisabled());
+        eyes.open(driver,
+                eyesConfig.getAppName() + "-" +
+                        config.getProperty(Config.PLATFORM).toUpperCase() + "-" +
+                        eyesConfig.getEnvironmentName(),
+                method.getDeclaringClass().getSimpleName() + "-" + method.getName(),
+                eyesConfig.getViewportSize());
+    }
+
+    @AfterMethod
+    public void closeVisualTest(ITestResult iTestResult, ITestContext context) {
+        LOGGER.info("@AfterMethod of VisualTest called: Waiting for visual validation results of test: " +
+                iTestResult.getName());
+        eyes.closeAsync();
+        TestResultsSummary allTestResults = runner.getAllTestResults(false);
+        TestResultContainer[] results = allTestResults.getAllResults();
+        LOGGER.info(String.format("Number of cross-browser tests run for test: %s: %d%n",
+                iTestResult.getName(), results.length));
+        boolean mismatchFound = false;
+        for (TestResultContainer eachResult : results) {
+            Throwable ex = results[0].getException();
+            TestResults testResult = eachResult.getTestResults();
+            mismatchFound = handleTestResults(testResult) || mismatchFound;
+        }
+        LOGGER.info(String.format("Overall Visual Validation failed? - %s%n", mismatchFound));
+        if (mismatchFound) {
+            String failingMessage = "Test: " + iTestResult.getName() + " has visual differences";
+            LOGGER.info(failingMessage);
+            //TODO - fail the test if there is a visual difference found\
```

Comment:
In case of visual differences, failing the test, or simply logging as an error message should be a configurable option.

Reply:
Added capability to fail the test in case visual differences found based on the property value (FAIL_TEST_WHEN_DIFFERENCE_FOUND) set in applittols.properties file